modules/processing/html_scrap.py

```diff
@@ -0,0 +1,106 @@
+import base64
+import binascii
+import logging
+import os
+import urllib.parse
+from typing import Optional
+
+from urlextract import URLExtract
+
+from lib.cuckoo.common.abstracts import Processing
+
+url_whitelist = (
+    'aadcdn.msftauth.net',
+    'https://aadcdn.msauth.net',
+    'https://aadcdn.msftauth.net',
+    'https://login.live.com',
+    'https://login.microsoftonline.com',
+    'https://outlook.office365.com',
+    'https://privacy.microsoft.com',
+    'https://www.microsoft.com',
+    'https://support.mozilla.org',
+    'http://www.w3.org',
+    'aadcdn.msauth.net',
+    'https://ajax.googleapis.com',
+    'https://code.jquery.com',
+    'https://fonts.googleapis.com',
+    'https://maxcdn.bootstrapcdn.com',
+    'https://kit.fontawesome.com',
+    'https://cdnjs.cloudflare.com',
+    'https://getbootstrap.com',
+    'https://use.fontawesome.com',
+    'https://www.office.com'
+)
+
+log = logging.getLogger(__name__)
+
+def try_base64_decode(text: str, validate: bool = True) -> Optional[bytes]:
+    try:
+        result = base64.b64decode(text, validate=validate)
+        return result
+    except binascii.Error:
+        return None
+
+
+def force_decode(text: str) -> Optional[str]:
+    base64_decoded_text = try_base64_decode(text, validate=True)
+    if base64_decoded_text:
+        if not base64_decoded_text.isascii():
+            return None
+
+        return force_decode(base64_decoded_text.decode())
```

Comment:
why the recursion? if it's needed, let's limit it to 3 level in case someone will base64 a base64